certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -151,15 +151,19 @@ func (p *DeployableByOlmCheck) isCSVReady(installedCSV string, operatorData Oper
 	go func() {
 		defer close(csvReadyDone)
```

Comment:
I think you need to pull the context, and the check for done, inside the goroutine. ctx.Done() does not actually terminate anything. It just informs that it's done. I have a thin grasp of it myself, and I'd be happy to discuss further.